c_common/models/live_packet_gather/src/live_packet_gather.c

```diff
@@ -450,64 +447,45 @@ static bool configure_sdp_msg(void) {
     temp_header |= config.payload_timestamp << PAYLOAD_IS_TIMESTAMP;
     temp_header |= config.packet_type << PACKET_TYPE;
 
-    header_len = 2;
-
     // pointers for AER packet header, prefix and data
-    void *temp_ptr;
+    sdp_msg_aer_data = &sdp_msg_aer_header[1];
     if (config.apply_prefix) {
-        // pointer to key prefix
-        sdp_msg_aer_key_prefix = &sdp_msg_aer_header[1];
-        temp_ptr = &sdp_msg_aer_header[2];
-        sdp_msg_aer_key_prefix[0] = (uint16_t) config.prefix;
-        header_len += 2;
-    } else {
-        sdp_msg_aer_key_prefix = NULL;
-        temp_ptr = &sdp_msg_aer_header[1];
+        // pointer to key prefix, so data is one half-word further ahead
+        write_short(sdp_msg_aer_header, 1, config.prefix);
+        sdp_msg_aer_data++;
     }
 
     if (config.payload_apply_prefix) {
-        sdp_msg_aer_payload_prefix = temp_ptr;
-        uint16_t *a = sdp_msg_aer_payload_prefix;
-
-        log_debug("temp_ptr: %08x", temp_ptr);
-        log_debug("a: %08x", a);
-
         // pointer to payload prefix
-        sdp_msg_aer_payload_prefix = temp_ptr;
+        sdp_msg_aer_payload_prefix = sdp_msg_aer_data;
 
         if (!HAVE_WIDE_LOAD(config.packet_type)) {
-            //16 bit payload prefix
-            temp_ptr = &a[1];
-            header_len += 2;
+            //16 bit payload prefix; advance data position by one half word
+            sdp_msg_aer_data++;
             if (!config.payload_timestamp) {
                 // add payload prefix as required - not a timestamp
-                a[0] = config.payload_prefix;
+                write_short(sdp_msg_aer_payload_prefix, 0, config.payload_prefix);
             }
-            log_debug("16 bit - temp_ptr: %08x", temp_ptr);
         } else {
-            //32 bit payload prefix
-            temp_ptr = &a[2];
-            header_len += 4;
+            //32 bit payload prefix; advance data position by two half words
+            sdp_msg_aer_data += 2;
             if (!config.payload_timestamp) {
                 // add payload prefix as required - not a timestamp
-                a[0] = CLAMP16(config.payload_prefix);
-                a[1] = CLAMP16(config.payload_prefix >> 16);
+                write_word(sdp_msg_aer_payload_prefix, 0, config.payload_prefix);
             }
-            log_debug("32 bit - temp_ptr: %08x", temp_ptr);
         }
     } else {
         sdp_msg_aer_payload_prefix = NULL;
     }
 
-    // pointer to write data
-    sdp_msg_aer_data = temp_ptr;
+    // compute header length
+    header_len = (sdp_msg_aer_data - sdp_msg_aer_header) * sizeof(uint16_t);
```

Comment:
We know that the data starts straight after the header, and we've kept track of where the header starts and the data starts, so we might as well just use those pointers to compute the header length directly rather than trying to calculate them as we go along.

Reply:
yea